Basic-Car-Maintenance/Shared/MainView/ViewModels/MainTabViewModel.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  MainTabViewModel.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Omar Hegazy on 13/10/2023.
+//
+
+import Foundation
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+import SwiftData
+
+@Observable
+class MainTabViewModel {
+    @MainActor var alert: AlertItem?
+    
+    /// Update the UI once a new alert is sent
+    func listenToAlertsUpdates(ignoring acknowledgedAlerts: [String]) {
+        
+        var query = Firestore
+            .firestore()
+            .collection(FirestoreCollection.alerts)
+            .whereField(FirestoreField.isOn, isEqualTo: true)
+            .limit(to: 1)
+        
+        if !acknowledgedAlerts.isEmpty {
+            query = query
+                .whereField(FirestoreField.id, notIn: acknowledgedAlerts)
+        }
+        
+        query.addSnapshotListener { [weak self] snapshot, error in
```

Comment:
Maybe realtime isn't really the right word. It's "realtime" as in faster than the App Store haha and sent from a backend

So yeah if you could change it to just a single fetch on app launch, that'd be great!

Otherwise someone could be doing something and they would suddenly get this alert, interrupt them, they'd dismiss it, not read it, and right now, have no way to get back that message